pkg/config/engine/containerd/config_v1.go

```diff
@@ -82,6 +90,14 @@ func (c *ConfigV1) AddRuntime(name string, path string, setAsDefault bool, confi
 		}
 	}
 
+	// If the systemd cgroup driver is required, and no other setting is applied, we set SystemdCgroup.
+	if c.CgroupDriver == cgroupDriverSystemd {
+		valueSet := config.GetPath([]string{"plugins", "cri", "containerd", "runtimes", name, "options", "SystemdCgroup"})
+		if valueSet == nil {
+			config.SetPath([]string{"plugins", "cri", "containerd", "runtimes", name, "options", "SystemdCgroup"}, true)
+		}
+	}
```

Comment:
Can we instead specifically look to see if `SystemdCgroup` is set, and if not, then set it appropriately (i.e. either true or false) depending on the value of `c.CgroupDriver`.

Reply:
So like this:
```
valueSet := config.GetPath([]string{"plugins", "cri", "containerd", "runtimes", name, "options", "SystemdCgroup"})
if valueSet == nil && c.CgroupDriver == cgroupDriverSystemd {
	config.SetPath([]string{"plugins", "cri", "containerd", "runtimes", name, "options", "SystemdCgroup"}, true)
}
if valueSet == nil && c.CgroupDriver == cgroupDriverCgroupFS {
	config.SetPath([]string{"plugins", "cri", "containerd", "runtimes", name, "options", "SystemdCgroup"}, false)
}

Reply:
I have moved the systemd-specifics to #578 to keep this PR more focussed.